packages/push/src/instruction/common/stack_depth.rs

```diff
@@ -0,0 +1,194 @@
+use std::marker::PhantomData;
+
+use crate::{
+    error::MapInstructionError,
+    instruction::{Instruction, instruction_error::PushInstructionError},
+    push_vm::HasStack,
+};
+
+/// An instruction that pushes a integer value onto the int stack
+/// that is the number of elements in the stack of type `T`.
+///
+/// # Inputs
+///
+/// The `StackDepth` instruction doesn't take any inputs from
+/// the stacks, but does compute the number of elements on the
+/// stack of type `T`.
+///
+/// # Behavior
+///
+/// The `StackDepth` instruction pushes a value onto the int
+/// stack that is the number of items currently on the stack
+/// of type `T` prior to the execution of this instruction.
+///
+/// If the type `T` is `int`, this will push the size of the `int`
+/// stack prior to the execution of this instruction onto the `int` stack,
+/// causing it to grow by one. If the `int`
+/// stack was empty, it would push the value 0 onto that stack.
+/// Otherwise the stack of type `T` is unchanged.
+///
+/// If the int stack is full before this instruction is performed, then
+/// a fatal [`StackError::Overflow`](crate::push_vm::stack::StackError::Overflow) is returned,
+/// terminating the running of the program.
+///
+/// If the size of the stack exceeds the maximum value of an
+/// `i64`, then `i64::MAX` will be pushed onto the int stack. This would only
+/// happen if there are 2^63 items on `Stack<T>`, which probably should
+/// not be allowed to happen since that is almost certainly the result
+/// of an infinite loop or similar and we should use `.with_max_stack_size()`
```

Comment:
might be worth to document on `.with_max_stack_size` that this instruction can "soft"-break if we set it too high.

Reply:
I added that in `generate_builder.rs`; see what you think.

Reply:
Sorry – those commits are now pushed.